src/renderer/src/routes/pages/BuyAndSellTokens/context.tsx

```diff
@@ -0,0 +1,76 @@
+import { createContext, Dispatch, ReactNode, useContext, useMemo, useState } from 'react'
+import { GateFiLangEnum, GateFiThemeType } from '@gatefi/js-sdk'
+import { useCurrencySelector } from '@renderer/hooks/useSettingsSelector'
+import { TAvailableCurrency } from '@shared/@types/store'
+
+const merchantId = import.meta.env.VITE_UNLIMIT_MERCHANT_ID
+const sellTokensIframeUrl = import.meta.env.VITE_UNLIMIT_SELL_TOKENS_IFRAME_URL
+
+type BuyAndSellTokensContextData = {
+  sellTokensIframeUrl: string
+  merchantId: string
+  externalId: string
+  currencyLabel: TAvailableCurrency
+  lang: GateFiLangEnum
+  theme: GateFiThemeType
+  hideBrand: boolean
+  isConfigured: boolean
+  isBuyTokensIframeLoading: boolean
+  setIsBuyTokensIframeLoading: Dispatch<boolean>
+  isSellTokensIframeLoading: boolean
+  setIsSellTokensIframeLoading: Dispatch<boolean>
+  hasSellTokensIframeError: boolean
+  setHasSellTokensIframeError: Dispatch<boolean>
+  isAccountsSideOpened: boolean
+  toggleAccountsSide(): void
+}
+
+const BuyAndSellTokensContext = createContext({} as BuyAndSellTokensContextData)
+
+type TProps = {
+  children: ReactNode
+}
+
+export const BuyAndSellTokensProvider = ({ children }: TProps) => {
```

Comment:
There is no need to create a context for this. Context is the same as RootStore is used in cases where you have to share states in different places of the application, that is not the case here. It makes it much more difficult to understand the flow of the page.

I think you should create different components for each page(buy/sell) and it one has its states. Create a shared place will turn these flows harder to understand.

Call me, so i can explain a better way to do that.